app/services/order_invoice_generator.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class OrderInvoiceGenerator
+  attr_reader :order
```

Comment:
This is unnecessarily creating a public getter method, which we're only using internally (below in comparator and invoice_data). Those methods could have just accessed the instance var directly.

But it hardly seems worth changing :)